helix-core/src/syntax.rs

```diff
@@ -1770,7 +1772,12 @@ const CANCELLATION_CHECK_INTERVAL: usize = 100;
 
 /// Indicates which highlight should be applied to a region of source code.
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
-pub struct Highlight(pub usize);
+pub enum Highlight {
+    /// When we use this type of highlight, we index into the Theme's scopes to get the Style
+    Indexed(usize),
+    /// A custom color, not dependent on the theme. Represents (red, green, blue)
+    Rgb(u8, u8, u8),
```

Comment:
```suggestion
    /// A custom color, not dependent on the theme.
    Rgb {
        red: u8,
        green: u8,
        blue: u8,
    },
```
Just an idea but I'm myself unsure if this is a good idea.

Reply:
Good idea! I was initially going to go with this, but since `helix_view::graphics::Color` is represented like this:

```rs
    Rgb(u8, u8, u8),
```

I think it's better to keep it a tuple for consistency